classes/shop/Shop.php

```diff
@@ -977,6 +985,18 @@ public static function getContextShopID($null_value_without_multishop = false)
     }
 
     /**
+     * @return int
+     */
+    public function getContextualShopId()
```

Comment:
The diff is currently too huge to display all the changes, hence I cannot see where you call it.

But if this is in the src/ files, you should move it in the src/Adapter/shop/Context.php ? Or reuse then existing functions `getContextShopID()`? I'm afraid that having several function doing almost the same the same makes the code more difficult to maintain.